Code/Framework/AzCore/AzCore/Task/TaskGraphSystemComponent.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Console/IConsole.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Task/TaskGraphSystemComponent.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+// Create a cvar as a central location for experimentation with switching from the Job system to TaskGraph system.
+AZ_CVAR(bool, cl_useTaskGraph, false, nullptr, AZ::ConsoleFunctorFlags::Null, "Flag for use of TaskGraph (Note does not disable task graph system)");
+
+namespace AZ
+{
+    void TaskGraphSystemComponent::Activate()
+    {
+        AZ_Assert(m_taskExecutor == nullptr, "Error multiple activation of the TaskGraphSystemComponent");
+
+        if (Interface<UseTaskGraphInterface>::Get() == nullptr)
+        {
+            Interface<UseTaskGraphInterface>::Register(this);
+            m_taskExecutor = aznew TaskExecutor();
+            m_taskExecutor->SetInstance(m_taskExecutor);
```

Comment:
Is this correct? m_taskExecutor calling SetInstance on itself.

Reply:
SetInstance is a static functions so technically it could be TaskExecutor::SetInstance, I'm willing to update if you object to this form.

Reply:
If you decide to leave this as is just add a quick comment.

Reply:
I would prefer `TaskExecutor::SetInstance` also FWIW

Reply:
I'll update to `TaskExecutor::SetInstance`